executor/executor_test.go

```diff
@@ -7337,3 +7337,35 @@ func (s *testSuite) TestIssue15563(c *C) {
 	tk := testkit.NewTestKit(c, s.store)
 	tk.MustQuery("select distinct 0.7544678906163867 /  0.68234634;").Check(testkit.Rows("1.10569639842486251190"))
 }
+
+func (s *testSuite) TestStalenessTransaction(c *C) {
+	testcases := []struct {
+		name        string
+		sql         string
+		IsStaleness bool
+	}{
+		{
+			name:        "TimestampBoundReadTimestamp",
+			sql:         `START TRANSACTION READ ONLY WITH TIMESTAMP BOUND READ TIMESTAMP '2020-09-06 00:00:00';`,
+			IsStaleness: true,
+		},
+		{
+			name:        "TimestampBoundExactStaleness",
+			sql:         `START TRANSACTION READ ONLY WITH TIMESTAMP BOUND EXACT STALENESS '00:00:05';`,
+			IsStaleness: true,
+		},
+		{
+			name:        "begin",
+			sql:         "begin",
+			IsStaleness: false,
+		},
+	}
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	for _, testcase := range testcases {
+		c.Log(testcase.name)
+		tk.MustExec(testcase.sql)
+		c.Assert(tk.Se.GetSessionVars().TxnCtx.IsStaleness, Equals, testcase.IsStaleness)
```

Comment:
Check the timestamp or duration, please.

Reply:
updated.